Version 3.0: massive overhaul, with most code moved to CanastaBase repo

yaronkoren opened the pull request.

yaronkoren committed:
Delete _sources/canasta directory

yaronkoren committed:
Delete _sources directory

yaronkoren committed:
Update Dockerfile

yaronkoren committed:
Update VERSION

yaronkoren committed:
Delete main.cf

yaronkoren committed:
Delete hadolint.yaml

yaronkoren committed:
Delete images directory

yaronkoren committed:
Update RELEASE_NOTES.md

yaronkoren committed:
Create cirrus-search-maintenance.sh

yaronkoren committed:
Create contents.yaml

yaronkoren committed:
Create getSMWSettings.php

yaronkoren committed:
Create smw-maintenance.sh

yaronkoren committed:
Restore hadolint.yaml

yaronkoren committed:
Update Dockerfile

yaronkoren committed:
Re-delete hadolint.yaml

yaronkoren committed:
Update docker-image.yml

yaronkoren committed:
Update RELEASE_NOTES.md

yaronkoren committed:
Update VERSION

yaronkoren committed:
Delete .github/ISSUE_TEMPLATE directory

The main functionality (and main source of bugs, presumably) is now in CanastaBase - while the set of extensions and skins is mostly determined by what's in "Recommended revisions". So there's no real need for any of these issue templates here - and they may just confuse users.

jeffw16 commented on .github/ISSUE_TEMPLATE/bug_report.md:
Why was this deleted?

naresh-kumar-babu reviewed (approved):
Looks good and everything works on testing.

yaronkoren committed:
Update release date to today

jeffw16 commented on .github/workflows/docker-image.yml:
```diff
@@ -58,9 +52,9 @@ jobs:
-          MEDIAWIKI_VERSION=$(sed -nr 's/MW_CORE_VERSION\=([0-9\.]+)/\1/p' Dockerfile | sed "s/ \\\//" | sed "s/\t//")
-          # Extract MW major version (like 1.35)
+          # We don't compute the MW version here, we just hardcode it
+          MEDIAWIKI_VERSION=1.43.0
```
How can this be sure that it will be updated each time? Would it be possible to echo it from a file like MEDIAWIKI_VERSION?

jeffw16 replied:
In that case, it would be best to echo this from a MEDIAWIKI_VERSION file in the root, just like the VERSION file.

`.github` is a hidden directory and it won't be obvious to update it upon an update.

It's unfortunate there is no immediately obvious way to be absolutely sure as of now. I don't think it's worth spending additional time on making a foolproof way, but I think at the very least, hardcoding it in `docker-image.yml` could be improved upon with this MEDIAWIKI_VERSION file approach.

yaronkoren replied:
That's reasonable - feel free to make the change.

jeffw16 replied:
Would someone else be able to make the change please? I am swamped.

yaronkoren replied:
I don't know, but I'd say if you can't do it yourself, you should undo this request, or whatever the terminology is... this would be a nice feature to have, but doesn't feel like a showstopper to me.

jeffw16 replied:
Ah wait, I have an idea. Why not just add back this env var into the Canasta (not Canasta Base) Dockerfile? Easy, doesn't harm anything else, and no need to write any new code.

yaronkoren replied:
Feel free to do that instead, then... it sounds like you're not completely swamped any more, which is great to hear.

jeffw16 commented:
@naresh-kumar-babu What’s up with the re-approval? It doesn’t override my request for changes. Please don’t try to undermine my requests for changes/clarification. I made them blocking for a reason.

yaronkoren commented:
I asked him to re-approve, because I didn't realize at the time that you had blocked this patch.

jeffw16 commented on .github/ISSUE_TEMPLATE/support_request.md:
```diff
@@ -1,43 +0,0 @@
----
```
Are we not doing support requests anymore? What will our users do when they make such requests anyway?

yaronkoren replied:
Just full out a blank issue, I suppose - I think it's a reasonable interface, becuase there's a good chance that the real issue, or missing functionality, will be in CanastaBase.

jeffw16 replied:
Removing the structure for people who still need the support request seems like a step backwards. Why remove it? It wasn’t doing any harm to anybody, but removing it is harmful to the volunteers who will have to waste time asking for info that could’ve already been provided had the template still existed.

yaronkoren replied:
I always thought these issue templates were a little too detailed, which I suppose in some small sense counts as harmful...   but the big thing, in my opinion, is that it's directing users to the wrong place, since presumably CanastaBase is where their issues should go.

jeffw16 replied:
But it was literally removed from there too. One of them needs to get it back.

yaronkoren replied:
Well, I did remove some of them, but that's a separate issue. (And I explained my reasoning there.) CanastaBase still has two templates, one for bugs and one for feature requests, which I think is a good number to have.

tosfos replied:
I recommend a compromise, keeping the structure but with a note that says something like "Filling out all fields is optional. But if important information is missing, it is less likely that anyone will take action."

jeffw16 commented:
I have no further concerns with the PR. Once these concerns are addressed, I'll approve this PR.

Well done team! Bravo and hats off to you!

yaronkoren committed:
"Recommended Revisions" is now capitalized

It's no longer "Recommended revisions". The old URL still works, but it's better to be fully correct.

yaronkoren committed:
Update RELEASE_NOTES.md

jeffw16 reviewed (approved):
Approving under duress

yaronkoren committed:
Update RELEASE_NOTES.md

yaronkoren committed:
Update Recommended Revisions URL (newer Cargo version)
